### PR TITLE
Adding the `hydrate` flag to the client methods

### PR DESCRIPTION
**Pull request by bcdurak**
## Describe changes

I have added the `hydrate` flag to the `get` and `list` methods in the client.  Fixed the `get_by_name_or_id_or_prefix` helper function as well.

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [X] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [X] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

## Summary by CodeRabbit

- **New Features**
	- Introduced a 'hydrate' option to various data retrieval methods, allowing users to choose whether to include metadata in their responses.
	
- **Enhancements**
	- Improved data fetching methods by adding a new parameter for better control over returned data.

### CODE REVIEW

**Comment by strickvl** [comment]
@coderabbitai review

File: src/zenml/zen_stores/zen_store_interface.py [hunk 1]
```diff
@@ -1355,13 +1368,17 @@ def get_service_account(
...
+        filter_model: ServiceAccountFilter,
+        hydrate: bool = True,
     ) -> Page[ServiceAccountResponse]:
         """List all service accounts.
 
         Args:
             filter_model: All filter parameters including pagination
                 params.
+            hydrate: Flag deciding whether to hydrate the output model(s)
+                by including metadata fields in the response.
```
[the remark this answers is not in the document]
Comment (strickvl) [reply src/zenml/zen_stores/zen_store_interface.py:1381]: @bcdurak Is there a special reason why this behaves differently?

**Review by strickvl — APPROVED** [review]
LGTM, though the coderabbit comment maybe needs addressing to be sure why the default setting is different in that specific point.

File: src/zenml/zen_stores/zen_store_interface.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bcdurak) [reply src/zenml/zen_stores/zen_store_interface.py:1381]: Good catch! It's a mistake from my side. Will fix it right away.

**Pull request merged**